### ISSUE TITLE
Not clear how to create nested components

### ISSUE THREAD

**Issue by zsid** [ueberdosis/tiptap#1484]
**Part of the documentation?**
https://www.tiptap.dev/api/nodes/table
https://www.tiptap.dev/guide/custom-extensions
https://www.tiptap.dev/guide/node-views
https://www.tiptap.dev/api/schema

**Really helpful parts**
It is helpful to see high level but building something yourself is really difficult

**Hard to understand, missing or misleading**
It is hard to understand doing your own extension from scratch or extending a current one.

**Additional context**
A guide on how to build a more complex extension from scratch will be helpful (i.e. build a Table, explaining what each method does and how it parses it to HTML, dom etc). For example, I want to have 
```
<div class="container">
  <p class=?>Editable content</p>
  <img class=? /> - your uploaded image comes here, either url or base64 - not concerned with that implementation at the moment
</div>
```
Or even simpler how to have a two column like the below and can edit text as you like.
```
<div class="hello">
   <p>Editable part</p>
   <p>Editable part 2</p>
</div>
```
Or another one:
```
<div className="hello">
   <p>Editable part</p>
   <table>Table goes here with columns which is again editable</table>
</div>
```

So basically it is a composition of components and functionality that someone might need.

Keep up the good work and thank you!

**Comment by hanspagel** [comment]
Thanks for the suggestion! I don't really know where to start with this issue. I hear your feedback, but I think it needs multiple solutions to improve that. I’m already in the process of restructuring and extending the documentation in my mind, so I’ll take this with me, but I don’t have any particular thing to solve that issue so I’m closing this for now. Thank you! ✌️
